Manual VMS update merge:

marcelino-a opened the pull request:
1. Refactoring of computation of vms stabilization parameter (tau)
2. Updated VMS edge bubble definition
3. Heuristic correction to tau for overlapping sectors (edge bubble support)
4.  Used stabilization parameter (tau) for jump norm of error when using vms

kuberry commented on physics/Compadre_ReactionDiffusion_Operator.cpp:
```diff
@@ -575,20 +596,20 @@ void ReactionDiffusionPhysics::initialize() {
         
-        //could parallelize outer two loops
+	// Allocate array of tau^(a)_s for all s, a
+        auto num_edges = _ndim_requested + 1; //MA: true for triangles and tetrahedra only
```
Drop this line and use num_sides in following line.

kuberry commented on examples/Compadre_ReactionDiffusion.cpp:
```diff
@@ -23,6 +23,13 @@
 #include <Compadre_ReactionDiffusion_BoundaryConditions.hpp>
 #include <Compadre_ReactionDiffusion_Operator.hpp>
 
+// ####################################
```
Drop

kuberry replied:
Add an assertion that it doesn't work for constant basis functions due to calculation of cubature_degree.

kuberry commented on physics/Compadre_ReactionDiffusion_Operator.cpp:
```diff
@@ -465,19 +479,44 @@ void ReactionDiffusionPhysics::initialize() {
+        for (int i=0; i<num_element_sides; ++i) {
+    	    for (int k=0; k<size_basis_T6_edge_values_1; ++k) {
+                double b1 = basis_T6_edge_values[i][k]; //values at SIDE cubature points
+    	        basis_T6_edge_values_pow[i][k] = bub_coeff * std::pow(b1, bub_pow) + (1.0 - bub_coeff) * b1;
```
Add comment of `heuristic` somewhere

kuberry commented:
Hi @marcelino-a. I made some changes to remove some commented out code and add some of the comments that we talked about. Would you mind pulling these changes and testing again to make sure it didn't break anything?

marcelino-a commented:
Thanks, Paul! I will pull and check the changes

marcelino-a commented:
Hi @kuberry, I pulled your changes. I also added some other changes of my own, and checked that everything is working:
1.I removed the variable "vmsBCfactor" from physics/Compadre_ReactionDiffusion_Sources.cpp and examples/Compadre_ReactionDiffusion.cpp because it was a feature used for testing that is no longer needed.
2. In physics/Compadre_ReactionDiffusion_Operator.cpp, in the block that determines cubature_degree for VMS, I made the logic a bit more explicit, as we had discussed.
3. I verified that VMSDG still gives optimal convergence (used variational_convergence.py, --solution=sine, checked for order=1,2,3,4 and 4 meshes)

kuberry commented:
Looks great! Thanks @marcelino-a
